imap_processing/swapi/l2/swapi_l2.py

```diff
@@ -0,0 +1,105 @@
+"""SWAPI L2 processing module."""
+
+import logging
+
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.swapi.swapi_utils import SWAPIAPID
+
+logger = logging.getLogger(__name__)
+
+
+TIME_PER_BIN = 0.167  # seconds
+
+
+def swapi_l2(l1_dataset: xr.Dataset, data_version: str) -> xr.Dataset:
+    """
+    Produce science data to L2.
+
+    To process science data to L2, we need to:
+    - convert counts to rates. This is done by dividing the counts by the
+        TIME_PER_BIN time. TIME_PER_BIN is the exposure time per energy bin which is
+        obtained by dividing the time for one complete sweep
+        (12 s, coarse + fine sweep) by the total energy steps (72),
+        i.e., TIME_PER_BIN = 12/72 = 0.167 s. This will be constant.
+
+    - update uncertainty. Calculate new uncertainty value using
+        SWP_PCEM_ERR data from level one and divide by TIME_PER_BIN. Eg.
+            SWP_PCEM_UNC = SWP_PCEM_ERR / TIME_PER_BIN
+        Do the same for SCEM and COIN data.
+
+    Parameters
+    ----------
+    l1_dataset : xarray.Dataset
+        The L1 data input.
+    data_version : str
+        Version of the data product being created.
+
+    Returns
+    -------
+    data : xarray.Dataset
+        Processed data to L2.
+    """
+    apid = int(l1_dataset.attrs["Apid"])
+    # Right now, we don't process any other apid besides science packet
+    # to L2
+    if apid != SWAPIAPID.SWP_SCI:
+        raise ValueError(f"APID {apid} is not supported for L2 processing")
+
+    # Load the CDF attributes
+    cdf_manager = ImapCdfAttributes()
+    cdf_manager.add_instrument_global_attrs("swapi")
+    cdf_manager.load_variable_attributes("imap_swapi_variable_attrs.yaml")
+
+    # Copy over only certain variables from L1 to L2 dataset
+    l1_data_keys = [
+        "epoch",
+        "energy",
+        "energy_label",
+        "swp_pcem_flags",
+        "swp_scem_flags",
+        "swp_coin_flags",
+    ]
+    l2_dataset = l1_dataset[l1_data_keys].copy()
+    # Copy over attributes
+    l2_dataset.attrs = l1_dataset.attrs
```

Comment:
Nice, I like this approach of keeping only what you need first rather than deleting what you don't. Thanks for updating it.

I don't think we need the copy, but can just use the view into it?
```suggestion
    l2_dataset = l1_dataset[l1_data_keys]
```
The attributes should already be brought along as well I thought.

Reply:
yeah, that worked. I was having issue before and I think it was because of `.copy()`. Thanks!